### PR TITLE
Bug 1601489 - Add locale to client_info section

### PR DESCRIPTION
**Pull request by brizental**
Checklist for reviewer:

- [x] Commits should reference a bug or github issue, if relevant
- [x] Scan the PR and verify that no changes (particularly to `.circleci/config.yml`) will cause environment variables (particularly credentials) to be exposed in test logs
- [x] Trigger the `integration` CI test by pushing this revision [as discussed in the README](https://github.com/mozilla-services/mozilla-pipeline-schemas#packaging-and-integration-tests-optional)

For glean changes:
- [x] Update `include/glean/CHANGELOG.md`

### CODE REVIEW

**Comment by brizental** [comment]
I was unsure if I should say this is a required field. Although it will be sent in every ping after https://github.com/mozilla/glean/pull/633 is merged, I wasn't sure adding it to the required fields list wouldn't break stuff.

**Comment by jklukas** [comment]
There are some steps that need to be run locally in order to get everything in place, which is non-obvious the first time you need to make changes in this repo. You'll need to follow a subset of the instructions in https://github.com/mozilla-services/mozilla-pipeline-schemas#adding-a-new-schema

> I was unsure if I should say this is a required field. Although it will be sent in every ping after mozilla/glean#633 is merged, I wasn't sure adding it to the required fields list wouldn't break stuff.

It's probably dangerous to make it required, at least for now. I expect we'll have pings coming in from clients on older glean versions without the field for some time. I'd recommend adding a new test message including this locale field under `validation/glean` to confirm the new schema accepts it and that the new schema doesn't reject existing pings without the field.

**Comment by brizental** [comment]
>  I'd recommend adding a new test message including this locale field under validation/glean to confirm the new schema accepts it and that the new schema doesn't reject existing pings without the field.

I added a test to check that having `locale` in the `client_info` doesn't fail. All the other tests don't have it, so they already validate that *not* having it doesn't fail.

**Comment by jklukas** [comment]
Actually, looks like we should update the include/glean/CHANGELOG.md as part of this PR.

**Comment by brizental** [comment]
> Actually, looks like we should update the include/glean/CHANGELOG.md as part of this PR.

Done. Let me know if I should merge this :)

File: templates/include/glean/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Version 1
 
+- Added `locale` to `client_info` section.
```
Comment (jklukas) [inline templates/include/glean/CHANGELOG.md]:
```suggestion
- Added `locale` to `client_info` section. See [Bug 1601489](https://bugzilla.mozilla.org/show_bug.cgi?id=1601489).
```

**Comment by jklukas** [comment]
One more suggestion to add a link to the bug in the changelog, and then feel free to merge.

**Comment by brizental** [comment]
> One more suggestion to add a link to the bug in the changelog, and then feel free to merge.

Made the changes. Sadly I don't have write access to this repo :(

**Pull request merged**